pkg/mutation/mutators/assign_mutator.go

```diff
@@ -278,7 +271,7 @@ func hasMetadataRoot(path *parser.Path) bool {
 
 // checkKeyNotChanged does not allow to change the key field of
 // a list element. A path like foo[name: bar].name is rejected
-func checkKeyNotChanged(p *parser.Path, assignName string) error {
+func checkKeyNotChanged(p parser.Path, assignName string) error {
 	if len(p.Nodes) == 0 || p.Nodes == nil {
```

Comment:
why do we need to check for `p.Nodes == nil` here?

Reply:
Good point, that should be false if len(p.Nodes) > 0

Reply:
Good catch!